org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/AbstractXbaseCompiler.java

```diff
@@ -656,11 +656,24 @@ protected LightweightTypeReference getTypeForVariableDeclaration(XExpression exp
 	 * If an expression has side effects this method must return true for it.
 	 * @param expr the checked expression
 	 * @param b the appendable which represents the current compiler state
+	 * @deprecated use {@link #isVariableDeclarationRequired(XExpression, ITreeAppendable, boolean)} instead.
 	 */
+	@Deprecated
 	protected boolean isVariableDeclarationRequired(XExpression expr, ITreeAppendable b) {
```

Comment:
If we make this method final, we will force overriding clients to take action, when recompiling their source code.
Which is important, because their overridden implementation will not be called anymore.

I checked github and there are around six overrides of isVariableDeclarationRequired that not only delegate to super. None of them seem to be on the Eclipse train, so they should be able to fix it.